[quant][fix][fx][graphmode] Fix qconfig setting for fused modules

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71254

Summary:
when we configure linear and relu with the same qconfig, we currently have utility functions to also
generate a qconfig for the fused linear relu module, but this code is not called in correct order before
which resulted in unexpected behaviors. This PR fixes the issue. Please see test case for more details.
(Test case is from Supriya)

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_fused_module_qat_swap

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33558321](https://our.internmc.facebook.com/intern/diff/D33558321)

jerryzh168 committed:
[quant][fix][fx][graphmode] Fix qconfig setting for fused modules

Summary:
when we configure linear and relu with the same qconfig, we currently have utility functions to also
generate a qconfig for the fused linear relu module, but this code is not called in correct order before
which resulted in unexpected behaviors. This PR fixes the issue. Please see test case for more details.
(Test case is from Supriya)

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_fused_module_qat_swap

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33558321).

supriyar commented on test/quantization/fx/test_quantize_fx.py:
```diff
@@ -496,6 +496,53 @@ def forward(self, x, y):
                 self.assertTrue(is_match(modules, n, pattern))
 
+    def test_fused_module_qat_swap(self):
+        class Linear(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.w = torch.ones(5, 5)
+                self.b = torch.zeros(5)
+
+            def forward(self, x):
+                return torch.nn.functional.linear(x, self.w, self.b)
+
```
I don't think this is used in the model

jerryzh168 replied:
oh OK, will remove

jerryzh168 committed:
Update on "[quant][fix][fx][graphmode] Fix qconfig setting for fused modules"


Summary:
when we configure linear and relu with the same qconfig, we currently have utility functions to also
generate a qconfig for the fused linear relu module, but this code is not called in correct order before
which resulted in unexpected behaviors. This PR fixes the issue. Please see test case for more details.
(Test case is from Supriya)

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_fused_module_qat_swap

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33558321](https://our.internmc.facebook.com/intern/diff/D33558321)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33558321).

jerryzh168 committed:
Update on "[quant][fix][fx][graphmode] Fix qconfig setting for fused modules"


Summary:
when we configure linear and relu with the same qconfig, we currently have utility functions to also
generate a qconfig for the fused linear relu module, but this code is not called in correct order before
which resulted in unexpected behaviors. This PR fixes the issue. Please see test case for more details.
(Test case is from Supriya)

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_fused_module_qat_swap

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33558321](https://our.internmc.facebook.com/intern/diff/D33558321)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix][fx][graphmode] Fix qconfig setting for fused modules"


Summary:
when we configure linear and relu with the same qconfig, we currently have utility functions to also
generate a qconfig for the fused linear relu module, but this code is not called in correct order before
which resulted in unexpected behaviors. This PR fixes the issue. Please see test case for more details.
(Test case is from Supriya)

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_fused_module_qat_swap

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33558321](https://our.internmc.facebook.com/intern/diff/D33558321)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33558321).

jerryzh168 committed:
Update on "[quant][fix][fx][graphmode] Fix qconfig setting for fused modules"


Summary:
when we configure linear and relu with the same qconfig, we currently have utility functions to also
generate a qconfig for the fused linear relu module, but this code is not called in correct order before
which resulted in unexpected behaviors. This PR fixes the issue. Please see test case for more details.
(Test case is from Supriya)

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_fused_module_qat_swap

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33558321](https://our.internmc.facebook.com/intern/diff/D33558321)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33558321).
